src/database/schema/BaseDbSchema.ts

```diff
@@ -2,7 +2,7 @@ import {SQLiteDatabase} from 'react-native-sqlite-storage';
 
 interface BaseDbSchema {
   save(db: SQLiteDatabase): Promise<void>;
-
+  saveIfNotExist(db: SQLiteDatabase): Promise<void>;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new method `saveIfNotExist` is introduced in the `BaseDbSchema` interface. Ensure that this method is implemented in all classes that implement this interface to avoid any runtime errors. If it's an optional method, consider marking it as such using the `?` symbol.
```diff
-  saveIfNotExist(db: SQLiteDatabase): Promise<void>;
+  saveIfNotExist?(db: SQLiteDatabase): Promise<void>;
```
This change will make `saveIfNotExist` an optional method, and classes implementing `BaseDbSchema` won't be forced to provide an implementation for it unless necessary.


<!-- This is an auto-generated comment by OSS CodeRabbit -->